src/util.rs

```diff
@@ -419,3 +419,165 @@ fn get_network_interfaces_impl() -> Vec<NetworkInterface> {
 
     vec
 }
+
+
+/// Perform a 16 bits words addition of an array of bytes
+pub fn sum_16_bit_words(data: &[u8]) -> u32 {
+    let mut sum = 0;
+    let length = data.len() - 1;
+    let mut i = 0;
+    while i < length {
+        let word = (data[i] as u32) << 8 | data[i + 1] as u32;
+        sum = sum + word;
+        i = i + 2;
+    }
+    if data.len() % 2 == 1 {
+        sum = sum + ((data[length] as u32) << 8);
+    }
+    sum
+}
+
+
+/// Calculates rfc1071 checksum value
+pub fn rfc1071_checksum(packet: &[u8], pseudo_header_packet: Option<&[u8]>) -> u16 {
+    let mut sum = 0;
+    if let Some(pseudo_header) = pseudo_header_packet {
+        sum = sum_16_bit_words(pseudo_header);
+    }
+    sum += sum_16_bit_words(packet);
+    while sum >> 16 != 0 {
+        sum = (sum >> 16) + (sum & 0xFFFF);
+    }
+    sum as u16 ^ 0xffff
+}
+
+#[test]
+fn checksum_even_simple() {
+    let data = vec![
+        0x03, 0x01, 0x00, 0x00, 0x12, 0x34, 0x56, 0x78,
+        0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08];
+    //   03 01
+    // + 00 00
+    // + 12 34
+    // + 56 78
+    // + 01 02
+    // + 03 04
+    // + 05 06
+    // + 07 08
+    //  +1 +2
+    // --------
+    //   7b c1 = 0111 1011 1100 0001
+    //
+    // => checksum is 1000 0100 0011 1110
+    //                   8    4    3    e
+    assert_eq!(rfc1071_checksum(&data[..], None), 0x843e);
+}
+
+#[test]
+fn checksum_even_with_carry() {
+    let data = vec![
```

Comment:
I'd rather these `vec![]`s be static arrays instead, so `let data = [...];` - it doesn't really matter for test cases, but is more consistent with that other tests do.

The styling on these functions is a little weird. Perhaps, given the comments below, it might be better to write them as:
```rust
let data = [
    0x30, 0x10, //    30 10
    0x00, 0x00, // + 00 00
    ...
];
```
That said, I'm not sure repeating the numbers in comments is particularly necessary.